src/utilities/timing.jl

```diff
@@ -0,0 +1,154 @@
+
+
+mutable struct BudgetedTimer
+    callback::Any
+    
+    target_delta_time::Float64
+    min_sleep::Float64
+    budget::Float64
+    last_time::UInt64
+    
+    running::Bool
+    task::Union{Nothing, Task}
+
+    function BudgetedTimer(callback, delta_time::Float64, running::Bool, task::Union{Nothing, Task}, min_sleep = 0.015)
+        return new(callback, delta_time, min_sleep, 0.0, time_ns(), running, task)
+    end
```

Comment:
I tried timing `sleep(0.0001)` here but it ended up being inconsistent. Most are 0.015s for me, but some are <0.005s and I assume async tasks and GC can also prolong them. So I just switched to a default value instead.

Reply:
Since I wasn't sure anymore I did some testing on what effect a wrong `min_sleep` has. Doesn't seem like there are any significant changes:
![Screenshot 2024-07-25 134212](https://github.com/user-attachments/assets/efb65596-3c18-4c0d-bfcb-10ce9b152d65)
![Screenshot 2024-07-25 134657](https://github.com/user-attachments/assets/160d3732-723f-4f5a-811e-2202e391db57)